### PR TITLE
Add a unique identifier to each feature

### PR DESCRIPTION
**Pull request by brian-avery**
This adds a unique, machine readable identifier to features. This can be used, among other things, for linking features to annotations.

@zhlsunshine @therealmitchconnors PTAL

### CODE REVIEW

**Comment by brian-avery** [comment]
Currently, this is just numeric, however, if area is necessary, we could also support an identifier of the format `areaID.featureID`. 

One con with doing this as an object property is I don't believe jsonschema supports verifying uniqueness. If I were to change it to be a key `id`:`object`, then we could. Could verify the uniqueness with either an additional check outside of the schema or updating all of the features as mentioned, but there are quite a few PRs in flight that I think we should get reviewed first.

File: features.yaml [hunk 1]
```diff
@@ -1,299 +1,344 @@
 features:
   - name: "Protocols:HTTP1.1/HTTP2/gRPC/TCP"
+    id: 1
```
Comment (therealmitchconnors) [inline features.yaml:3]: What would you think of making the id meaningful to humans as well?  something like basic-protocols?

Then, when we want to refer to this feature elsewhere (for instance in @zhlsunshine's API maturity annotations), the reference is slightly readable, where an integer is not.

Comment (brian-avery) [reply features.yaml:3]: Sure. I'm fine with that. Will the API maturity annotations look up the feature name? The name is rendered to Istio.io so may be useful for consistency in things like istioctl.

**Comment by therealmitchconnors** [comment]
Yes, that is the idea.

**Review by zhlsunshine — APPROVED** [review]

**Comment by zhlsunshine** [comment]
Hi @brian-avery ,  LGTM,  I have approved it. Let's see the other's, thank you for your effort on this PR! ^_^

**Review by therealmitchconnors — APPROVED** [review]

File: features.yaml [hunk 2]
```diff
@@ -1,299 +1,344 @@
...
     area: Traffic Management
   - name: "SNI (multiple certs) at ingress"
+    id: "traffic.multicertingresssni"
     level:
       checklist: ""
       maturity: Stable
       nextExpectedPromotion: ""
     area: Traffic Management
   - name: "Locality load balancing"
+    id: "traffic.localityloadbalancing"
```
Comment (howardjohn) [inline features.yaml:59]: We now have two mappings of "features" - the other one being here https://github.com/istio/istio/blob/809c761526f43e05853d1b8f389449a41e7c4d27/pkg/test/framework/features/features.yaml#L36 and doesn't have the same identifiers. it seems like we should have one

Comment (brian-avery) [reply features.yaml:59]: Yeah, the IDs should probably be the same.

**Review by jasonwzm — APPROVED** [review]

File: features.yaml (hunk 2 above)
Comment (brian-avery) [reply features.yaml]: Talking to @therealmitchconnors, it sounds like either the other feature IDs will go away, or they will be updated to match the ones here so I think that using the ones here is fine. 

There's also not a one-to-one mapping with what's in the other features.yaml and what's here.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**